



# Determining Success and Failure Factors for Startups
***

## Introduction

The purpose of this project is to find out the success and failure factors behind startup companies. To that end, I am going to investigate a dataset provided by [Metric.am](https://metric.am/), containing inforation about 472 startups and their status: *'Success'* or *'Failed'*. 

## Methodology

For the sake of this project, we will consider two classes of startups: *failed* and *successful*. As already mentioned above, we are interested in the factors behind the success or failure of the startup. If we tried to map the problem to the data, we would be interested in the importance of each feature or attribute while predicting the status of the company. Thus, it would be most meaningful and comfortable to solve the problem using a model which would provide us with this information. 

A model that seems suitable under the mentioned constraints is Logstic Regression. It is a classification algorithm, so it can trained to predict the status of the company. It also associates weights with idividual features, which we can use to determine the importance of the feature, as well as point out if its effect is positive or negative.

So this point onwards, we are going to pursue the following steps:
- [Retrieve the data](#Retrieving-data)
- [Explore the data](#Exploring-data)
- [Prepare the data]()
- [Build the classifier]()
    - [Split the data into train-test sets]()
    - [Build the model]()
    - [Train the model]()
- [Test the accuracy]()
- [Analyze the calculated weights]()
- [Report the conclusions](#Report)

***
## Retrieving data

First of all, let's download the data into a *.csv* file. I will use [wget](https://www.gnu.org/software/wget/) to download raw data from the [GitHub repository](https://github.com/Metricam/Internship_tasks/tree/master/Startup_Success).

In [617]:
!wget -O data.csv https://raw.githubusercontent.com/Metricam/Internship_tasks/master/Startup_Success/data.csv

--2020-05-19 20:22:05--  https://raw.githubusercontent.com/Metricam/Internship_tasks/master/Startup_Success/data.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.16.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.16.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 315563 (308K) [text/plain]
Saving to: ‘data.csv’

100%[======================================>] 315,563     --.-K/s   in 0.02s   

2020-05-19 20:22:06 (14.2 MB/s) - ‘data.csv’ saved [315563/315563]



We are also provided a *dictionary* dataset, which explains some of the features provided in the main dataset. Let's get that too.

In [618]:
!wget -O dictionary.csv https://raw.githubusercontent.com/Metricam/Internship_tasks/master/Startup_Success/dictionary.csv

--2020-05-19 20:22:07--  https://raw.githubusercontent.com/Metricam/Internship_tasks/master/Startup_Success/dictionary.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.16.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.16.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 5183 (5.1K) [text/plain]
Saving to: ‘dictionary.csv’

100%[======================================>] 5,183       --.-K/s   in 0s      

2020-05-19 20:22:07 (42.5 MB/s) - ‘dictionary.csv’ saved [5183/5183]



***
## Exploring data

Now let's understand the data. For that we will need to import some libraries. (Notice, you might need to install the libraries. If that's the case, un-comment the first four lines and then run the block.)

In [619]:
# import sys
# !{sys.executable} -m pip install numpy
# !{sys.executable} -m pip install pandas
# !{sys.executable} -m pip install matplotlib

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn import preprocessing
import re

### About dataset

What are the attributes?

In [620]:
data_dict = pd.read_csv('dictionary.csv')
pd.options.display.max_rows = None
pd.options.display.max_columns = None
display(data_dict)

Variable  \
0                                         Company_Name   
1                             Dependent-Company Status   
2                                     year of founding   
3                              Age of company in years   
4                              Internet Activity Score   
5                 Short Description of company profile   
6                                  Industry of company   
7                           Focus functions of company   
8                                            Investors   
9                                       Employee Count   
10                          Employees count MoM change   
11                             Has the team size grown   
12                                  Est. Founding Date   
13                                   Last Funding Date   
14                                 Last Funding Amount   
15                                  Country of company   
16                                Continent of company   
17                         Number of Investors in Seed   
18              Number of Investors in Angel and or VC   
19                               Number of Co-founders   
20                               Number of of advisors   
21                         Team size Senior leadership   
22                             Team size all employees   
23   Presence of a top angel or venture fund in pre...   
24                       Number of of repeat investors   
25                   Number of  Sales Support material   
26                             Worked in top companies   
27    Average size of companies worked for in the past   
28             Have been part of startups in the past?   
29   Have been part of successful startups in the p...   
30          Was he or she partner in Big 5 consulting?   
31                              Consulting experience?   
32                         Product or service company?   
33        Catering to product/service across verticals   
34                    Focus on private or public data?   
35                             Focus on consumer data?   
36            Focus on structured or unstructured data   
37                         Subscription based business   
38             Cloud or platform based serive/product?   
39                              Local or global player   
40                 Linear or Non-linear business model   
41   Capital intensive business e.g. e-commerce, En...   
42                   Number of  of Partners of company   
43                        Crowdsourcing based business   
44                         Crowdfunding based business   
45                     Machine Learning based business   
46                       Predictive Analytics business   
47                           Speech analytics business   
48                     Prescriptive analytics business   
49                                   Big Data Business   
50         Cross-Channel Analytics/ marketing channels   
51   Owns data or not? (monetization of data) e.g. ...   
52   Is the company an aggregator/market place? e.g...   
53   Online or offline venture - physical location ...   
54                                 B2C or B2B venture?   
55   Top forums like 'Tech crunch' or 'Venture beat...   
56   Average Years of experience for founder and co...   
57                           Exposure across the globe   
58              Breadth of experience across verticals   
59                                   Highest education   
60                                  Years of education   
61                 Specialization of highest education   
62                   Relevance of education to venture   
63                  Relevance of experience to venture   
64          Degree from a Tier 1 or Tier 2 university?   
65                     Renowned in professional circle   
66         Experience in selling and building products   
67             Experience in Fortune 100 organizations   
68             Experience in Fortune 500 org

Let's take a look at our data. 

In [621]:
#df = pd.read_csv('data.csv')
#df.head()

The default encoder in read_csv() function is *UTF-8*, however, as we can see from the above error, the data in the *data.csv* file in not UTF-8 encoded. Let's try several other encodings to see if any of them succeeds to decode the data. Here are some: *latin1*, *iso-8859-1*, *cp1252*.

In [622]:
df = pd.read_csv('data.csv', encoding='latin1')
df.head()
#display(df)

Company_Name Dependent-Company Status year of founding  \
0     Company1                  Success          No Info   
1     Company2                  Success             2011   
2     Company3                  Success             2011   
3     Company4                  Success             2009   
4     Company5                  Success             2010   

  Age of company in years  Internet Activity Score  \
0                 No Info                     -1.0   
1                       3                    125.0   
2                       3                    455.0   
3                       5                    -99.0   
4                       4                    496.0   

     Short Description of company profile  \
0                      Video distribution   
1                                     NaN   
2                Event Data Analytics API   
3  The most advanced analytics for mobile   
4   The Location-Based Marketing Platform   

                              Industry of company Focus functions of company  \
0                                             NaN                  operation   
1          Market Research|Marketing|Crowdfunding           Marketing, sales   
2  Analytics|Cloud Computing|Software Development                 operations   
3                                Mobile|Analytics          Marketing & Sales   
4         Analytics|Marketing|Enterprise Software          Marketing & Sales   

                                           Investors  Employee Count  \
0  KPCB Holdings|Draper Fisher Jurvetson (DFJ)|Kl...             3.0   
1                                                NaN             NaN   
2  TechStars|Streamlined Ventures|Amplify Partner...            14.0   
3  Michael Birch|Max Levchin|Sequoia Capital|Keit...            45.0   
4  DFJ Frontier|Draper Nexus Ventures|Gil Elbaz|A...            39.0   

   Employees count MoM change Has the team size grown Est. Founding Date  \
0                         0.0                      No                NaN   
1                         NaN                      No                NaN   
2                         0.0                      No          12/1/2011   
3                        10.0                      No          6/20/2009   
4                         3.0                      No           4/1/2010   

  Last Funding Date  Last Funding Amount Country of company  \
0         5/26/2013             450000.0      United States   
1               NaN                  NaN      United States   
2        10/23/2013            2350000.0      United States   
3         5/10/2012           10250000.0      United States   
4        12/11/2013            5500000.0      United States   

  Continent of company Number of Investors in Seed  \
0        North America                           2   
1        North America                           5   
2        North America                          15   
3        North America                           6   
4        North America                           7   

  Number of Investors in Angel and or VC  Number of Co-founders  \
0                                      0                      1   
1                                      0                      2   
2                                      0                      3   
3                                      0                      2   
4                                      0                      1   

   Number of of advisors  Team size Senior leadership Team size all employees  \
0                      2                            2                      15   
1                      0                            4                      20   
2                      0                            7                      10   
3                      0                            4                      50   
4                      1                            8                      40   

  Presence of a top angel or venture fund in previous round of investment  \
0  

Lucky for us, *latin1* was the real encoding!

***
## Preparing data

Now let's prepare our data for classification and separate the features from the labels. 

**Let's print the shape of our dataset before cleaning and hope that we'll not destroy it comletely after cleaning. :D**

In [623]:
print("Shape before cleaning: ", df.shape)

Shape before cleaning:  (472, 116)


### Choosing the feature set...

First of all, it seems like following columns don't give too much value for our purpose: *'Company_Name', 'Short Description of company profile', 'Last Funding Date', 'Est. Founding Date', 'Age of company in years', 'Gartner hype cycle stage', 'Time to maturity of technology (in years)'*, so let's drop them!

Also, the status column should not be in the feature set.

In [624]:
features = df.drop(['Dependent-Company Status', 'Company_Name', 'Short Description of company profile', 
                    'Last Funding Date', 'Est. Founding Date', 'Age of company in years', 
                    'Gartner hype cycle stage', 'Time to maturity of technology (in years)'], axis=1)
features.head()

year of founding  Internet Activity Score  \
0          No Info                     -1.0   
1             2011                    125.0   
2             2011                    455.0   
3             2009                    -99.0   
4             2010                    496.0   

                              Industry of company Focus functions of company  \
0                                             NaN                  operation   
1          Market Research|Marketing|Crowdfunding           Marketing, sales   
2  Analytics|Cloud Computing|Software Development                 operations   
3                                Mobile|Analytics          Marketing & Sales   
4         Analytics|Marketing|Enterprise Software          Marketing & Sales   

                                           Investors  Employee Count  \
0  KPCB Holdings|Draper Fisher Jurvetson (DFJ)|Kl...             3.0   
1                                                NaN             NaN   
2  TechStars|Streamlined Ventures|Amplify Partner...            14.0   
3  Michael Birch|Max Levchin|Sequoia Capital|Keit...            45.0   
4  DFJ Frontier|Draper Nexus Ventures|Gil Elbaz|A...            39.0   

   Employees count MoM change Has the team size grown  Last Funding Amount  \
0                         0.0                      No             450000.0   
1                         NaN                      No                  NaN   
2                         0.0                      No            2350000.0   
3                        10.0                      No           10250000.0   
4                         3.0                      No            5500000.0   

  Country of company Continent of company Number of Investors in Seed  \
0      United States        North America                           2   
1      United States        North America                           5   
2      United States        North America                          15   
3      United States        North America                           6   
4      United States        North America                           7   

  Number of Investors in Angel and or VC  Number of Co-founders  \
0                                      0                      1   
1                                      0                      2   
2                                      0                      3   
3                                      0                      2   
4                                      0                      1   

   Number of of advisors  Team size Senior leadership Team size all employees  \
0                      2                            2                      15   
1                      0                            4                      20   
2                      0                            7                      10   
3                      0                            4                      50   
4                      1                            8                      40   

  Presence of a top angel or venture fund in previous round of investment  \
0                                                Yes                        
1                                                 No                        
2                                                 No                        
3                                                Yes                        
4                                                 No                        

  Number of of repeat investors Number of  Sales Support material  \
0                             4                           Nothing   
1                             0                            medium   
2                             0                               low   
3                             0                               low   
4                             0                              high   

  Worked in top companies Average size of companies worked for in the past  \
0                      No                          

In [625]:
statuses = df['Dependent-Company Status']
statuses.head()

0    Success
1    Success
2    Success
3    Success
4    Success
Name: Dependent-Company Status, dtype: object

### Handling NaN values...

Before we rush to drop the rows that have at least one NaN value, let's see which features have the most NaN values. Maybe the presence of those features is less important compared to the amount of data we'll lose...

In [626]:
nan_df = pd.DataFrame(features.isna().sum().sort_values(ascending=False))
print(nan_df[nan_df[0] != 0].count())
nan_df[nan_df[0] != 0]

0    12
dtype: int64


0
Employees count MoM change           205
Employee Count                       166
Last Funding Amount                  160
Investors                            140
Industry of company                  124
Specialization of highest education   97
Industry trend in investing           82
Continent of company                  71
Country of company                    71
Internet Activity Score               65
Has the team size grown               50
Focus functions of company            30

The overall picture is not that bad, from 116 features, only 12 have NaN values, the rest are either fully filled in or the NaN values are expressed verbally, like 'no info'. Let's now quickly go over the list and see if it makes sense to keep these features.

- **'Employees count MoM change'** makes nearly the half of the dataset drop, so let's just get rid of that feature, as I don't even know what MoM is (Ministry of Manpower?).
- **'Employee Count'** who doesn't have this record? It seems like such a  primitive question! Anyways, this is an important data, and, sadly, we have to drop 166 lines!
- **'Last Funding Amount'** let's just drop this column.
- **'Investors'** this one is different, my intuition is that the rows that don't have investors have NaN values, so instead of dropping them, I'm going to substitute those with empty strings. Later, when I will make this variable binary, those rows will just have 0s for all the Investors. 
- **'Industry of company'** is pretty important in my opinion, so I'm going to drop the rows that have NaN.
- **'Specialization of highest education'** is a crazy field: some have meaningful answers and some have things like _"PhD"_ or _"INDUSTRI"_ or _"computers"_. Plus, this field is not directly connected to the success, in my opinion, the connection between the field of the startup and the specialization is what matters. We could filter all the fields out from specialization in complex ways and then compare with the fields of the startup, but if we take a quick look at the dataset, we'll see that the answers vary a lot and maybe it's best in this case to just discard this column and not over-engineer.
- The rest are fairly important and don't have drastic results, so we'll just drom the NaN rows.

In [627]:
# Drop the discussed columns.
features = features.drop(['Employees count MoM change', 'Last Funding Amount', 
                          'Specialization of highest education'], axis=1)

# Sustitute NaNs with empty strings.
features['Investors'] = features['Investors'].apply(lambda x: x if isinstance(x, str) else '')

# Drop the rest of the rows.
features = features.dropna()

features.head()

year of founding  Internet Activity Score  \
2             2011                    455.0   
3             2009                    -99.0   
4             2010                    496.0   
5             2010                    106.0   
6             2011                     39.0   

                              Industry of company Focus functions of company  \
2  Analytics|Cloud Computing|Software Development                 operations   
3                                Mobile|Analytics          Marketing & Sales   
4         Analytics|Marketing|Enterprise Software          Marketing & Sales   
5                    Food & Beverages|Hospitality                  analytics   
6                                       Analytics                   Research   

                                           Investors  Employee Count  \
2  TechStars|Streamlined Ventures|Amplify Partner...            14.0   
3  Michael Birch|Max Levchin|Sequoia Capital|Keit...            45.0   
4  DFJ Frontier|Draper Nexus Ventures|Gil Elbaz|A...            39.0   
5  Pritzker Group Venture Capital|Excelerate Labs...            14.0   
6  Plug & Play Ventures|Correlation Ventures|Cros...             7.0   

  Has the team size grown Country of company Continent of company  \
2                      No      United States        North America   
3                      No      United States        North America   
4                      No      United States        North America   
5                      No      United States        North America   
6                      No      United States        North America   

  Number of Investors in Seed Number of Investors in Angel and or VC  \
2                          15                                      0   
3                           6                                      0   
4                           7                                      0   
5                           2                                      0   
6                           7                                      0   

   Number of Co-founders  Number of of advisors  Team size Senior leadership  \
2                      3                      0                            7   
3                      2                      0                            4   
4                      1                      1                            8   
5                      4                      0                            4   
6                      2                      9                            2   

  Team size all employees  \
2                      10   
3                      50   
4                      40   
5                      14   
6                      15   

  Presence of a top angel or venture fund in previous round of investment  \
2                                                 No                        
3                                                Yes                        
4                                                 No                        
5                                                 No                        
6                                                 No                        

  Number of of repeat investors Number of  Sales Support material  \
2                             0                               low   
3                             0                               low   
4                             0                              high   
5                             2                            medium   
6                             4                            medium   

  Worked in top companies Average size of companies worked for in the past  \
2                     Yes                                           Medium   
3                      No                                            Large   
4                      No                                            Small   
5                      No                                           Medium   
6                      No        

In [628]:
features.shape

(239, 105)

In [629]:
display(features)

year of founding  Internet Activity Score  \
2               2011                    455.0   
3               2009                    -99.0   
4               2010                    496.0   
5               2010                    106.0   
6               2011                     39.0   
7               2010                    139.0   
8               2011                    306.0   
9               2013                     53.0   
10              2011                    762.0   
12              2010                    115.0   
13              2008                    277.0   
14              2010                    242.0   
15              2008                    533.0   
16              2010                    -81.0   
17              2009                    471.0   
19              2010                     60.0   
20              2010                     70.0   
22              2012                    245.0   
24              2012                    373.0   
25              2011                     61.0   
26              2009                    155.0   
29              2012                    281.0   
30              2012                    421.0   
31              2008                    -91.0   
32              2010                    983.0   
33              2011                     68.0   
35              2012                     75.0   
36              2012                    140.0   
37              2012                     46.0   
38              2010                     33.0   
39              2010                     56.0   
40              2012                     89.0   
41              2009                     69.0   
46              2011                    295.0   
48              2000                    160.0   
50              2011                    805.0   
51              2009                   -129.0   
57              2012                    272.0   
59              2012                    124.0   
60              2009                    409.0   
62              2012                    388.0   
63              2012                    315.0   
66              2009                    157.0   
70              2009                     43.0   
72              2012                    217.0   
74              2013                    181.0   
75              2012                    -34.0   
76              2012                     14.0   
77              2012                    398.0   
78              2010                    372.0   
80              2010                    263.0   
82              2011                     42.0   
83              2011                    138.0   
84              2011                    401.0   
86              2010                   -103.0   
87              2009                     -2.0   
89              2013                    763.0   
90              2010                    -13.0   
91              2011                    293.0   
92              2010                    126.0   
94              2002                    123.0   
95              2011                     53.0   
99              2012                    220.0   
101             2012                     -2.0   
102             2010                    -34.0   
104             2009                    265.0   
106             2008                     98.0   
107             2011                    338.0   
109             2010                     16.0   
110             2010                    361.0   
113             2012                    161.0   
114             2009                    230.0   
116             2011                    528.0   
117             2009                    512.0   
118             2005                     78.0   
119             2008                    150.0   
120             2012                    129.0   
121             2010                     -4.0   
122             2011                     29.0   
123             2010                    -68.0   
124             2011               

### Handling not NaN empty values...

Although we removed a bunch of rows with NaN values, there are still some records that in principle are not NaN but have the same effect. As those values can be anything, I am going to manually look into the dataset to detect them. To, nevertheless, ease the job, I'll look into the set of unique values of each column. 

In [630]:
col_sets = []
for c in features.columns:
    col_sets.append(c)
    col_sets.append(features[c].unique())
    
#col_sets
col_sets[20:26] # Just showing a sample batch.

['Number of Investors in Angel and or VC',
 array(['0', '2', '3', '6', '1', 'No Info', '4', '5', '7', '9', '8'],
       dtype=object),
 'Number of Co-founders',
 array([3, 2, 1, 4, 0, 5, 7, 6]),
 'Number of of advisors',
 array([ 0,  1,  9,  4,  8,  2,  6,  3,  7,  5, 11])]

 **Based on my obervations, here are the NaN value strings:**

- 'No Info' 
<!-- 'Number of Investors in Seed', 'Number of Investors in Angel and or VC', 'Presence of a top angel or venture fund in previous round of investment', 'year of founding', 'Team size all employees', 'Presence of a top angel or venture fund in previous round of investment', 'Subscription based business', 'Local or global player', 'Linear or Non-linear business model', 'Capital intensive business e.g. e-commerce, Engineering products and operations can also cause a business to be capital intensive', 'Number of  of Partners of company', 'Online or offline venture - physical location based business or online venture?', 'Exposure across the globe', 'Renowned in professional circle', 'Breadth of experience across verticals', 'Highest education', 'Years of education', 'Relevance of education to venture', 'Relevance of experience to venture', 'Degree from a Tier 1 or Tier 2 university?', 'Employees per year of company existence', 'Time to 1st investment (in months)', 'Avg time to investment - average across all rounds, measured from previous investment', 'Team Composition score', 'Pricing Strategy', 'Time to market service or product', 'Employee benefits and salary structures', 'Long term relationship with other founders', 'Client Reputation', 'Solutions offered', 'Invested through global incubation competitions?', 'Survival through recession, based on existence of the company through recession times' -->
- '\\\\' 
- 'N'
- 'unknown amount'
 

Let's see what are the losses of cleaning in this case.

In [631]:
no_info_df = pd.DataFrame(features.apply(lambda x: (x == "No Info") | (x == "\\") 
                        | (x == "N") | (x == "unknown amount"))
                          .sum().sort_values(ascending=False))

print(no_info_df[no_info_df[0] != 0].count())
no_info_df[no_info_df[0] != 0]

0    36
dtype: int64


/opt/conda/envs/Python36/lib/python3.6/site-packages/pandas/core/ops.py:1649: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  result = method(y)


0
Employee benefits and salary structures             156
Client Reputation                                   103
Last round of funding received (in milionUSD)        43
Presence of a top angel or venture fund in prev...   36
google page rank of company website                  26
Invested through global incubation competitions?     21
Years of education                                   13
Highest education                                    13
Employees per year of company existence              13
Number of  of Partners of company                    11
Local or global player                                9
Relevance of education to venture                     9
Time to 1st investment (in months)                    9
Avg time to investment - average across all rou...    8
Capital intensive business e.g. e-commerce, Eng...    8
year of founding                                      7
Renowned in professional circle                       6
Linear or Non-linear business model                   6
Survival through recession, based on existence ...    6
Average size of companies worked for in the past      4
Degree from a Tier 1 or Tier 2 university?            3
Long term relationship with other founders            3
Team size all employees                               2
Focus functions of company                            2
Relevance of experience to venture                    2
Team Composition score                                2
Number of Investors in Seed                           2
Number of Investors in Angel and or VC                2
Pricing Strategy                                      1
Time to market service or product                     1
Breadth of experience across verticals                1
Subscription based business                           1
Exposure across the globe                             1
Online or offline venture - physical location b...    1
Solutions offered                                     1
Catering to product/service across verticals          1

Let's drop the first 3 columns, as they cause major data loss, and clear the rest.

In [632]:
features = features.drop(['Employee benefits and salary structures', 'Client Reputation', 
                          'Presence of a top angel or venture fund in previous round of investment'], axis=1)

In [633]:
for col in features.columns:
    features = features[(features[col] != "No Info") & (features[col] != "\\") 
                        & (features[col] != "N") & (features[col] != "unknown amount")]
    
print(features.shape)
features = features.reset_index(drop=True)
features.head()

(118, 102)


year of founding  Internet Activity Score  \
0             2011                    455.0   
1             2010                    496.0   
2             2010                    106.0   
3             2010                    139.0   
4             2011                    306.0   

                                 Industry of company  \
0     Analytics|Cloud Computing|Software Development   
1            Analytics|Marketing|Enterprise Software   
2                       Food & Beverages|Hospitality   
3  Cloud Computing|Network / Hosting / Infrastruc...   
4                         Analytics|Mobile|Marketing   

  Focus functions of company  \
0                 operations   
1          Marketing & Sales   
2                  analytics   
3                  Computing   
4                  Marketing   

                                           Investors  Employee Count  \
0  TechStars|Streamlined Ventures|Amplify Partner...            14.0   
1  DFJ Frontier|Draper Nexus Ventures|Gil Elbaz|A...            39.0   
2  Pritzker Group Venture Capital|Excelerate Labs...            14.0   
3  Norwest Venture Partners|Bessemer Venture Part...            29.0   
4  Promus Ventures|SoftTech VC|Costanoa Venture C...            16.0   

  Has the team size grown Country of company Continent of company  \
0                      No      United States        North America   
1                      No      United States        North America   
2                      No      United States        North America   
3                      No      United States        North America   
4                      No      United States        North America   

  Number of Investors in Seed Number of Investors in Angel and or VC  \
0                          15                                      0   
1                           7                                      0   
2                           2                                      0   
3                           0                                      0   
4                          13                                      0   

   Number of Co-founders  Number of of advisors  Team size Senior leadership  \
0                      3                      0                            7   
1                      1                      1                            8   
2                      4                      0                            4   
3                      3                      4                            4   
4                      2                      0                            2   

  Team size all employees Number of of repeat investors  \
0                      10                             0   
1                      40                             0   
2                      14                             2   
3                      40                             0   
4                      50                             0   

  Number of  Sales Support material Worked in top companies  \
0                               low                     Yes   
1                              high                      No   
2                            medium                      No   
3                            medium                      No   
4                            medium                     Yes   

  Average size of companies worked for in the past  \
0                                           Medium   
1                                            Small   
2                                           Medium   
3                                            Large   
4                                           Medium   

  Have been part of startups in the past?  \
0                                      No   
1                                      No   
2                                      No   
3                                     Yes   
4                                     Yes   

  Have been part of successful startups in the past?  \
0                                        

### Converting list values into binary variables...

Let's now convert the following values into lists, so that we can then make them binary variables.

 - *'Industry of company'*
 - *'Investors'*
 - *'Focus functions of company'*

In [634]:
features['Industry of company'] = features['Industry of company'].apply(lambda x: x.lower().strip().split('|'))
features['Investors'] = features['Investors'].apply(lambda x: x.lower().strip().split('|'))

def process_str(x) :
    str_list = re.split(',|&|\n|\|', x.lower())
    str_list = list(map(str.strip, str_list))
    return str_list

features['Focus functions of company'] = features['Focus functions of company'].apply(lambda x: process_str(x))

features.head()

year of founding  Internet Activity Score  \
0             2011                    455.0   
1             2010                    496.0   
2             2010                    106.0   
3             2010                    139.0   
4             2011                    306.0   

                                 Industry of company  \
0  [analytics, cloud computing, software developm...   
1        [analytics, marketing, enterprise software]   
2                    [food & beverages, hospitality]   
3  [cloud computing, network / hosting / infrastr...   
4                     [analytics, mobile, marketing]   

  Focus functions of company  \
0               [operations]   
1         [marketing, sales]   
2                [analytics]   
3                [computing]   
4                [marketing]   

                                           Investors  Employee Count  \
0  [techstars, streamlined ventures, amplify part...            14.0   
1  [dfj frontier, draper nexus ventures, gil elba...            39.0   
2  [pritzker group venture capital, excelerate la...            14.0   
3  [norwest venture partners, bessemer venture pa...            29.0   
4  [promus ventures, softtech vc, costanoa ventur...            16.0   

  Has the team size grown Country of company Continent of company  \
0                      No      United States        North America   
1                      No      United States        North America   
2                      No      United States        North America   
3                      No      United States        North America   
4                      No      United States        North America   

  Number of Investors in Seed Number of Investors in Angel and or VC  \
0                          15                                      0   
1                           7                                      0   
2                           2                                      0   
3                           0                                      0   
4                          13                                      0   

   Number of Co-founders  Number of of advisors  Team size Senior leadership  \
0                      3                      0                            7   
1                      1                      1                            8   
2                      4                      0                            4   
3                      3                      4                            4   
4                      2                      0                            2   

  Team size all employees Number of of repeat investors  \
0                      10                             0   
1                      40                             0   
2                      14                             2   
3                      40                             0   
4                      50                             0   

  Number of  Sales Support material Worked in top companies  \
0                               low                     Yes   
1                              high                      No   
2                            medium                      No   
3                            medium                      No   
4                            medium                     Yes   

  Average size of companies worked for in the past  \
0                                           Medium   
1                                            Small   
2                                           Medium   
3                                            Large   
4                                           Medium   

  Have been part of startups in the past?  \
0                                      No   
1                                      No   
2                                      No   
3                                     Yes   
4                                     Yes   

  Have been part of successful startups in the past?  \
0                                        

In [635]:
# For every row in the dataframe, iterate through the list of industries 
# and place a 1 into the corresponding column.
for index, row in features.iterrows():
    for industry in row['Industry of company']:
        features.at[index, industry] = 1
        
# Filling in the NaN values with 0 to show that a company doesn't belong to that column's industry.
features = features.fillna(0)

# Same for investors.
for index, row in features.iterrows():
    for investor in row['Investors']:
        features.at[index, investor] = 1

features = features.fillna(0)

# And for focus functions of company.
for index, row in features.iterrows():
    for focus in row['Focus functions of company']:
        features.at[index, focus] = 1
        
features = features.fillna(0)
        
features.head()

year of founding  Internet Activity Score  \
0             2011                    455.0   
1             2010                    496.0   
2             2010                    106.0   
3             2010                    139.0   
4             2011                    306.0   

                                 Industry of company  \
0  [analytics, cloud computing, software developm...   
1        [analytics, marketing, enterprise software]   
2                    [food & beverages, hospitality]   
3  [cloud computing, network / hosting / infrastr...   
4                     [analytics, mobile, marketing]   

  Focus functions of company  \
0               [operations]   
1         [marketing, sales]   
2                [analytics]   
3                [computing]   
4                [marketing]   

                                           Investors  Employee Count  \
0  [techstars, streamlined ventures, amplify part...            14.0   
1  [dfj frontier, draper nexus ventures, gil elba...            39.0   
2  [pritzker group venture capital, excelerate la...            14.0   
3  [norwest venture partners, bessemer venture pa...            29.0   
4  [promus ventures, softtech vc, costanoa ventur...            16.0   

  Has the team size grown Country of company Continent of company  \
0                      No      United States        North America   
1                      No      United States        North America   
2                      No      United States        North America   
3                      No      United States        North America   
4                      No      United States        North America   

  Number of Investors in Seed Number of Investors in Angel and or VC  \
0                          15                                      0   
1                           7                                      0   
2                           2                                      0   
3                           0                                      0   
4                          13                                      0   

   Number of Co-founders  Number of of advisors  Team size Senior leadership  \
0                      3                      0                            7   
1                      1                      1                            8   
2                      4                      0                            4   
3                      3                      4                            4   
4                      2                      0                            2   

  Team size all employees Number of of repeat investors  \
0                      10                             0   
1                      40                             0   
2                      14                             2   
3                      40                             0   
4                      50                             0   

  Number of  Sales Support material Worked in top companies  \
0                               low                     Yes   
1                              high                      No   
2                            medium                      No   
3                            medium                      No   
4                            medium                     Yes   

  Average size of companies worked for in the past  \
0                                           Medium   
1                                            Small   
2                                           Medium   
3                                            Large   
4                                           Medium   

  Have been part of startups in the past?  \
0                                      No   
1                                      No   
2                                      No   
3                                     Yes   
4                                     Yes   

  Have been part of successful startups in the past?  \
0                                        

And drop the original columns with lists.

In [636]:
features = features.drop('Industry of company', 1).drop('Investors', 1).drop('Focus functions of company', 1)
features.head()

year of founding  Internet Activity Score  Employee Count  \
0             2011                    455.0            14.0   
1             2010                    496.0            39.0   
2             2010                    106.0            14.0   
3             2010                    139.0            29.0   
4             2011                    306.0            16.0   

  Has the team size grown Country of company Continent of company  \
0                      No      United States        North America   
1                      No      United States        North America   
2                      No      United States        North America   
3                      No      United States        North America   
4                      No      United States        North America   

  Number of Investors in Seed Number of Investors in Angel and or VC  \
0                          15                                      0   
1                           7                                      0   
2                           2                                      0   
3                           0                                      0   
4                          13                                      0   

   Number of Co-founders  Number of of advisors  Team size Senior leadership  \
0                      3                      0                            7   
1                      1                      1                            8   
2                      4                      0                            4   
3                      3                      4                            4   
4                      2                      0                            2   

  Team size all employees Number of of repeat investors  \
0                      10                             0   
1                      40                             0   
2                      14                             2   
3                      40                             0   
4                      50                             0   

  Number of  Sales Support material Worked in top companies  \
0                               low                     Yes   
1                              high                      No   
2                            medium                      No   
3                            medium                      No   
4                            medium                     Yes   

  Average size of companies worked for in the past  \
0                                           Medium   
1                                            Small   
2                                           Medium   
3                                            Large   
4                                           Medium   

  Have been part of startups in the past?  \
0                                      No   
1                                      No   
2                                      No   
3                                     Yes   
4                                     Yes   

  Have been part of successful startups in the past?  \
0                                                 No   
1                                                 No   
2                                                 No   
3                                                Yes   
4                                                Yes   

  Was he or she partner in Big 5 consulting? Consulting experience?  \
0                                         No                     No   
1                                         No                     No   
2                                         No                     No   
3                                         No                     No   
4                                         No                     No   

  Product or service company? Catering to product/service across verticals  \
0                        Both                                          Yes   
1                     Product               

### Encoding categorical variables as numericals...

First, let's encode the statuses.

In [637]:
statuses.unique()

array(['Success', 'Failed'], dtype=object)

In [638]:
statuses.replace(to_replace=['Success', 'Failed'], value=[1,0], inplace=True)
statuses.head()

0    1
1    1
2    1
3    1
4    1
Name: Dependent-Company Status, dtype: int64

Let's consider answers having less than 20 unique values as categorical in our feature list.

In [639]:
cat_columns = []
cat_vals = []
for col in features.columns:
    if len(features[col].unique()) < 21:
        if isinstance(features[col][0], str):
            col_set = sorted(features[col].apply(lambda x: x.lower()).unique())
        else:
            col_set = sorted(features[col].unique())
        if not((len(col_set) == 2) and (col_set[0] == 0) and (col_set[1] == 1)):
            cat_columns.append(col)
            cat_vals.append(col_set)
        
categoricals = pd.DataFrame()
categoricals['Feature'] = cat_columns
categoricals['Categories'] = cat_vals

categoricals

Feature  \
0                                    year of founding   
1                             Has the team size grown   
2                                  Country of company   
3                                Continent of company   
4                         Number of Investors in Seed   
5              Number of Investors in Angel and or VC   
6                               Number of Co-founders   
7                               Number of of advisors   
8                         Team size Senior leadership   
9                       Number of of repeat investors   
10                  Number of  Sales Support material   
11                            Worked in top companies   
12   Average size of companies worked for in the past   
13            Have been part of startups in the past?   
14  Have been part of successful startups in the p...   
15         Was he or she partner in Big 5 consulting?   
16                             Consulting experience?   
17                        Product or service company?   
18       Catering to product/service across verticals   
19                   Focus on private or public data?   
20                            Focus on consumer data?   
21           Focus on structured or unstructured data   
22                        Subscription based business   
23            Cloud or platform based serive/product?   
24                             Local or global player   
25                Linear or Non-linear business model   
26  Capital intensive business e.g. e-commerce, En...   
27                  Number of  of Partners of company   
28                       Crowdsourcing based business   
29                        Crowdfunding based business   
30                    Machine Learning based business   
31                      Predictive Analytics business   
32                          Speech analytics business   
33                    Prescriptive analytics business   
34                                  Big Data Business   
35        Cross-Channel Analytics/ marketing channels   
36  Owns data or not? (monetization of data) e.g. ...   
37  Is the company an aggregator/market place? e.g...   
38  Online or offline venture - physical location ...   
39                                B2C or B2B venture?   
40  Top forums like 'Tech crunch' or 'Venture beat...   
41  Average Years of experience for founder and co...   
42                          Exposure across the globe   
43             Breadth of experience across verticals   
44                                  Highest education   
45                                 Years of education   
46                  Relevance of education to venture   
47                 Relevance of experience to venture   
48         Degree from a Tier 1 or Tier 2 university?   
49                    Renowned in professional circle   
50        Experience in selling and building products   
51            Experience in Fortune 100 organizations   
52            Experience in Fortune 500 organizations   
53           Experience in Fortune 1000 organizations   
54                          Top management similarity   
55                Number of  of Research publications   
56                             Team Composition score   
57                  Dificulty of Obtaining Work force   
58                                   Pricing Strategy   
59                                 Hyper localisation   
60                  Time to market service or product   
61         Long term relationship with other founders   
62  Proprietary or patent position (competitive po...   
63              Barriers of entry for the competitors   
64                                     Company awards   
65     Controversial history of founder or co founder   
66               Legal risk and intellectual property   
67  Technical proficiencies to analyse and interpr...   
68                                  Solutions offered   
69   Invested through global incubation competitions?   

Let's take out all the features that have two options and replace them with 0/1s. This includes yes/no, linear/non-linear, local/global, etc..

In [640]:
for row, col in zip(range(len(categoricals)), categoricals.Feature):
    if isinstance(features[col][0], str):
        col_set = categoricals['Categories'][row]
        if (len(col_set) == 2):
            features[col] = features[col].apply(lambda x: x.lower())
            features[col].replace(to_replace=col_set, value=[0,1], inplace=True)
            categoricals = categoricals[categoricals['Feature'] != col]
features.head()

year of founding  Internet Activity Score  Employee Count  \
0             2011                    455.0            14.0   
1             2010                    496.0            39.0   
2             2010                    106.0            14.0   
3             2010                    139.0            29.0   
4             2011                    306.0            16.0   

   Has the team size grown Country of company Continent of company  \
0                        0      United States        North America   
1                        0      United States        North America   
2                        0      United States        North America   
3                        0      United States        North America   
4                        0      United States        North America   

  Number of Investors in Seed Number of Investors in Angel and or VC  \
0                          15                                      0   
1                           7                                      0   
2                           2                                      0   
3                           0                                      0   
4                          13                                      0   

   Number of Co-founders  Number of of advisors  Team size Senior leadership  \
0                      3                      0                            7   
1                      1                      1                            8   
2                      4                      0                            4   
3                      3                      4                            4   
4                      2                      0                            2   

  Team size all employees Number of of repeat investors  \
0                      10                             0   
1                      40                             0   
2                      14                             2   
3                      40                             0   
4                      50                             0   

  Number of  Sales Support material  Worked in top companies  \
0                               low                        1   
1                              high                        0   
2                            medium                        0   
3                            medium                        0   
4                            medium                        1   

  Average size of companies worked for in the past  \
0                                           Medium   
1                                            Small   
2                                           Medium   
3                                            Large   
4                                           Medium   

   Have been part of startups in the past?  \
0                                        0   
1                                        0   
2                                        0   
3                                        1   
4                                        1   

   Have been part of successful startups in the past?  \
0                                                  0    
1                                                  0    
2                                                  0    
3                                                  1    
4                                                  1    

   Was he or she partner in Big 5 consulting?  Consulting experience?  \
0                                           0                       0   
1                                           0                       0   
2                                           0                       0   
3                                           0                       0   
4                                           0                       0   

  Product or service company?  Catering to product/service across verticals  \
0                        Both                                             1   
1     

In [641]:
categoricals = categoricals.reset_index(drop=True)
categoricals

Feature  \
0                                    year of founding   
1                                  Country of company   
2                                Continent of company   
3                         Number of Investors in Seed   
4              Number of Investors in Angel and or VC   
5                               Number of Co-founders   
6                               Number of of advisors   
7                         Team size Senior leadership   
8                       Number of of repeat investors   
9                   Number of  Sales Support material   
10   Average size of companies worked for in the past   
11                        Product or service company?   
12                   Focus on private or public data?   
13           Focus on structured or unstructured data   
14            Cloud or platform based serive/product?   
15                  Number of  of Partners of company   
16  Online or offline venture - physical location ...   
17  Top forums like 'Tech crunch' or 'Venture beat...   
18  Average Years of experience for founder and co...   
19             Breadth of experience across verticals   
20                                  Highest education   
21                                 Years of education   
22         Degree from a Tier 1 or Tier 2 university?   
23                    Renowned in professional circle   
24        Experience in selling and building products   
25                          Top management similarity   
26                Number of  of Research publications   
27                             Team Composition score   
28                  Dificulty of Obtaining Work force   
29                  Time to market service or product   
30     Controversial history of founder or co founder   
31                        Industry trend in investing   
32                       Disruptiveness of technology   
33                       Number of Direct competitors   
34  Survival through recession, based on existence...   
35                           Percent_skill_Consulting   
36                              Percent_skill_Finance   
37                                       Renown score   

                                           Categories  
0   [1999, 2000, 2002, 2004, 2005, 2006, 2007, 200...  
1   [belgium, bulgaria, czech republic, denmark, f...  
2                       [asia, europe, north america]  
3   [0, 1, 10, 13, 15, 17, 2, 22, 24, 3, 4, 5, 6, ...  
4                      [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]  
5                               [0, 1, 2, 3, 4, 5, 7]  
6                     [0, 1, 2, 3, 4, 5, 6, 7, 8, 11]  
7             [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]  
8                                 [0, 1, 10, 2, 3, 4]  
9                        [high, low, medium, nothing]  
10                             [large, medium, small]  
11                           [both, product, service]  
12                        [both, no, private, public]  
13  [both, no, not applicable, structured, unstruc...  
14                      [both, cloud, none, platform]  
15                                  [few, many, none]  
16                            [both, offline, online]  
17                          [high, low, medium, none]  
18                                [high, low, medium]  
19                                [high, low, medium]  
20                          [bachelors, masters, phd]  
21                                       [18, 21, 25]  
22                       [both, none, tier_1, tier_2]  
23  [146, 300, 324, 329, 367, 400, 416, 417, 439, ...  
24                          [high, low, medium, none]  
25                          [high, low, medium, none]  
26                                  [few, many, none]  
27                                [high, low, medium]  
28                                [high, low, medium]  
29                                [high, low, medium]  
30                                               [no]  
31                            

As we can see, the *'Controversial history of founder or co founder'* feature has a unique answer throughout the database, so let's just drop it, as it does not give us any information.

In [642]:
features = features.drop('Controversial history of founder or co founder', 1)
categoricals = categoricals[categoricals['Feature'] != 'Controversial history of founder or co founder']

categoricals

Feature  \
0                                    year of founding   
1                                  Country of company   
2                                Continent of company   
3                         Number of Investors in Seed   
4              Number of Investors in Angel and or VC   
5                               Number of Co-founders   
6                               Number of of advisors   
7                         Team size Senior leadership   
8                       Number of of repeat investors   
9                   Number of  Sales Support material   
10   Average size of companies worked for in the past   
11                        Product or service company?   
12                   Focus on private or public data?   
13           Focus on structured or unstructured data   
14            Cloud or platform based serive/product?   
15                  Number of  of Partners of company   
16  Online or offline venture - physical location ...   
17  Top forums like 'Tech crunch' or 'Venture beat...   
18  Average Years of experience for founder and co...   
19             Breadth of experience across verticals   
20                                  Highest education   
21                                 Years of education   
22         Degree from a Tier 1 or Tier 2 university?   
23                    Renowned in professional circle   
24        Experience in selling and building products   
25                          Top management similarity   
26                Number of  of Research publications   
27                             Team Composition score   
28                  Dificulty of Obtaining Work force   
29                  Time to market service or product   
31                        Industry trend in investing   
32                       Disruptiveness of technology   
33                       Number of Direct competitors   
34  Survival through recession, based on existence...   
35                           Percent_skill_Consulting   
36                              Percent_skill_Finance   
37                                       Renown score   

                                           Categories  
0   [1999, 2000, 2002, 2004, 2005, 2006, 2007, 200...  
1   [belgium, bulgaria, czech republic, denmark, f...  
2                       [asia, europe, north america]  
3   [0, 1, 10, 13, 15, 17, 2, 22, 24, 3, 4, 5, 6, ...  
4                      [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]  
5                               [0, 1, 2, 3, 4, 5, 7]  
6                     [0, 1, 2, 3, 4, 5, 6, 7, 8, 11]  
7             [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]  
8                                 [0, 1, 10, 2, 3, 4]  
9                        [high, low, medium, nothing]  
10                             [large, medium, small]  
11                           [both, product, service]  
12                        [both, no, private, public]  
13  [both, no, not applicable, structured, unstruc...  
14                      [both, cloud, none, platform]  
15                                  [few, many, none]  
16                            [both, offline, online]  
17                          [high, low, medium, none]  
18                                [high, low, medium]  
19                                [high, low, medium]  
20                          [bachelors, masters, phd]  
21                                       [18, 21, 25]  
22                       [both, none, tier_1, tier_2]  
23  [146, 300, 324, 329, 367, 400, 416, 417, 439, ...  
24                          [high, low, medium, none]  
25                          [high, low, medium, none]  
26                                  [few, many, none]  
27                                [high, low, medium]  
28                                [high, low, medium]  
29                                [high, low, medium]  
31                               [1.0, 2.0, 3.0, 4.0]  
32                                [high, low, medium]  
33       [0, 1, 10, 11, 19, 2, 

Let's now rule out the Features that don't seem feasible to be viewed as categoricals. These features are:

- 'Percent_skill_Consulting'
- 'Percent_skill_Finance'
- 'Renowned in professional circle'

The rest are better off as categoricals, as the dependency should not neccessarily be linear, and we can better analyze having those features as categoricals.

In [643]:
categoricals = categoricals[categoricals['Feature'] != 'Percent_skill_Consulting']
categoricals = categoricals[categoricals['Feature'] != 'Percent_skill_Finance']
categoricals = categoricals[categoricals['Feature'] != 'Renowned in professional circle']
categoricals = categoricals.reset_index(drop=True)

categoricals

Feature  \
0                                    year of founding   
1                                  Country of company   
2                                Continent of company   
3                         Number of Investors in Seed   
4              Number of Investors in Angel and or VC   
5                               Number of Co-founders   
6                               Number of of advisors   
7                         Team size Senior leadership   
8                       Number of of repeat investors   
9                   Number of  Sales Support material   
10   Average size of companies worked for in the past   
11                        Product or service company?   
12                   Focus on private or public data?   
13           Focus on structured or unstructured data   
14            Cloud or platform based serive/product?   
15                  Number of  of Partners of company   
16  Online or offline venture - physical location ...   
17  Top forums like 'Tech crunch' or 'Venture beat...   
18  Average Years of experience for founder and co...   
19             Breadth of experience across verticals   
20                                  Highest education   
21                                 Years of education   
22         Degree from a Tier 1 or Tier 2 university?   
23        Experience in selling and building products   
24                          Top management similarity   
25                Number of  of Research publications   
26                             Team Composition score   
27                  Dificulty of Obtaining Work force   
28                  Time to market service or product   
29                        Industry trend in investing   
30                       Disruptiveness of technology   
31                       Number of Direct competitors   
32  Survival through recession, based on existence...   
33                                       Renown score   

                                           Categories  
0   [1999, 2000, 2002, 2004, 2005, 2006, 2007, 200...  
1   [belgium, bulgaria, czech republic, denmark, f...  
2                       [asia, europe, north america]  
3   [0, 1, 10, 13, 15, 17, 2, 22, 24, 3, 4, 5, 6, ...  
4                      [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]  
5                               [0, 1, 2, 3, 4, 5, 7]  
6                     [0, 1, 2, 3, 4, 5, 6, 7, 8, 11]  
7             [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]  
8                                 [0, 1, 10, 2, 3, 4]  
9                        [high, low, medium, nothing]  
10                             [large, medium, small]  
11                           [both, product, service]  
12                        [both, no, private, public]  
13  [both, no, not applicable, structured, unstruc...  
14                      [both, cloud, none, platform]  
15                                  [few, many, none]  
16                            [both, offline, online]  
17                          [high, low, medium, none]  
18                                [high, low, medium]  
19                                [high, low, medium]  
20                          [bachelors, masters, phd]  
21                                       [18, 21, 25]  
22                       [both, none, tier_1, tier_2]  
23                          [high, low, medium, none]  
24                          [high, low, medium, none]  
25                                  [few, many, none]  
26                                [high, low, medium]  
27                                [high, low, medium]  
28                                [high, low, medium]  
29                               [1.0, 2.0, 3.0, 4.0]  
30                                [high, low, medium]  
31       [0, 1, 10, 11, 19, 2, 28, 3, 33, 4, 5, 6, 8]  
32                          [no, not applicable, yes]  
33                 [0, 1, 10, 2, 3, 4, 5, 6, 7, 8, 9]

Great! Now we are left only with the features that should be encoded as binary variables, like we did in case of lists. We just ahve to make sure that every column has the original feature name attached to the answer so that noting gets overwrote (e.g. *'both'*, *'none'*). 

In [644]:
for index, ignored in features.iterrows():
    for row, feature in zip(range(len(categoricals)), categoricals.Feature):
        for option in categoricals['Categories'][row]:
            col_name = feature + " " + str(option)
            features.at[index, col_name] = 1
        
features = features.fillna(0)

features.head()

year of founding  Internet Activity Score  Employee Count  \
0             2011                    455.0            14.0   
1             2010                    496.0            39.0   
2             2010                    106.0            14.0   
3             2010                    139.0            29.0   
4             2011                    306.0            16.0   

   Has the team size grown Country of company Continent of company  \
0                        0      United States        North America   
1                        0      United States        North America   
2                        0      United States        North America   
3                        0      United States        North America   
4                        0      United States        North America   

  Number of Investors in Seed Number of Investors in Angel and or VC  \
0                          15                                      0   
1                           7                                      0   
2                           2                                      0   
3                           0                                      0   
4                          13                                      0   

   Number of Co-founders  Number of of advisors  Team size Senior leadership  \
0                      3                      0                            7   
1                      1                      1                            8   
2                      4                      0                            4   
3                      3                      4                            4   
4                      2                      0                            2   

  Team size all employees Number of of repeat investors  \
0                      10                             0   
1                      40                             0   
2                      14                             2   
3                      40                             0   
4                      50                             0   

  Number of  Sales Support material  Worked in top companies  \
0                               low                        1   
1                              high                        0   
2                            medium                        0   
3                            medium                        0   
4                            medium                        1   

  Average size of companies worked for in the past  \
0                                           Medium   
1                                            Small   
2                                           Medium   
3                                            Large   
4                                           Medium   

   Have been part of startups in the past?  \
0                                        0   
1                                        0   
2                                        0   
3                                        1   
4                                        1   

   Have been part of successful startups in the past?  \
0                                                  0    
1                                                  0    
2                                                  0    
3                                                  1    
4                                                  1    

   Was he or she partner in Big 5 consulting?  Consulting experience?  \
0                                           0                       0   
1                                           0                       0   
2                                           0                       0   
3                                           0                       0   
4                                           0                       0   

  Product or service company?  Catering to product/service across verticals  \
0                        Both                                             1   
1     

Let's drop the original columns.

In [645]:
features = features.reset_index(drop=True)

for feature in categoricals['Feature']:
    features = features.drop(feature, 1)
    
features.head()

Internet Activity Score  Employee Count  Has the team size grown  \
0                    455.0            14.0                        0   
1                    496.0            39.0                        0   
2                    106.0            14.0                        0   
3                    139.0            29.0                        0   
4                    306.0            16.0                        0   

  Team size all employees  Worked in top companies  \
0                      10                        1   
1                      40                        0   
2                      14                        0   
3                      40                        0   
4                      50                        1   

   Have been part of startups in the past?  \
0                                        0   
1                                        0   
2                                        0   
3                                        1   
4                                        1   

   Have been part of successful startups in the past?  \
0                                                  0    
1                                                  0    
2                                                  0    
3                                                  1    
4                                                  1    

   Was he or she partner in Big 5 consulting?  Consulting experience?  \
0                                           0                       0   
1                                           0                       0   
2                                           0                       0   
3                                           0                       0   
4                                           0                       0   

   Catering to product/service across verticals  Focus on consumer data?  \
0                                             1                        1   
1                                             1                        1   
2                                             0                        1   
3                                             1                        0   
4                                             1                        1   

   Subscription based business  Local or global player  \
0                            1                       1   
1                            0                       1   
2                            0                       1   
3                            0                       1   
4                            0                       1   

   Linear or Non-linear business model  \
0                                    1   
1                                    1   
2                                    1   
3                                    1   
4                                    1   

   Capital intensive business e.g. e-commerce, Engineering products and operations can also cause a business to be capital intensive  \
0                                                  0                                                                                   
1                                                  1                                                                                   
2                                                  0                                                                                   
3                                                  0                                                                                   
4                                                  0                                                                                   

   Crowdsourcing based business  Crowdfunding based business  \
0                             0                            0   
1                             0                            0   
2                             0                            0   
3                             0                            0

Now let's make sure everything is a float type in our dataset.

In [646]:
for col in features.columns:
    try:
        features[col] = features[col].apply(lambda x: float(x))
    except:
        pass

features.head()

Internet Activity Score  Employee Count  Has the team size grown  \
0                    455.0            14.0                      0.0   
1                    496.0            39.0                      0.0   
2                    106.0            14.0                      0.0   
3                    139.0            29.0                      0.0   
4                    306.0            16.0                      0.0   

   Team size all employees  Worked in top companies  \
0                     10.0                      1.0   
1                     40.0                      0.0   
2                     14.0                      0.0   
3                     40.0                      0.0   
4                     50.0                      1.0   

   Have been part of startups in the past?  \
0                                      0.0   
1                                      0.0   
2                                      0.0   
3                                      1.0   
4                                      1.0   

   Have been part of successful startups in the past?  \
0                                                0.0    
1                                                0.0    
2                                                0.0    
3                                                1.0    
4                                                1.0    

   Was he or she partner in Big 5 consulting?  Consulting experience?  \
0                                         0.0                     0.0   
1                                         0.0                     0.0   
2                                         0.0                     0.0   
3                                         0.0                     0.0   
4                                         0.0                     0.0   

   Catering to product/service across verticals  Focus on consumer data?  \
0                                           1.0                      1.0   
1                                           1.0                      1.0   
2                                           0.0                      1.0   
3                                           1.0                      0.0   
4                                           1.0                      1.0   

   Subscription based business  Local or global player  \
0                          1.0                     1.0   
1                          0.0                     1.0   
2                          0.0                     1.0   
3                          0.0                     1.0   
4                          0.0                     1.0   

   Linear or Non-linear business model  \
0                                  1.0   
1                                  1.0   
2                                  1.0   
3                                  1.0   
4                                  1.0   

   Capital intensive business e.g. e-commerce, Engineering products and operations can also cause a business to be capital intensive  \
0                                                0.0                                                                                   
1                                                1.0                                                                                   
2                                                0.0                                                                                   
3                                                0.0                                                                                   
4                                                0.0                                                                                   

   Crowdsourcing based business  Crowdfunding based business  \
0                           0.0                          0.0   
1                           0.0                          0.0   
2                           0.0                          0.0   
3                           0.0                       

To be even surer, lets print the data types:

In [647]:
features.dtypes


Internet Activity Score                                                                                                              float64
Employee Count                                                                                                                       float64
Has the team size grown                                                                                                              float64
Team size all employees                                                                                                              float64
Worked in top companies                                                                                                              float64
Have been part of startups in the past?                                                                                              float64
Have been part of successful startups in the past?                                                                                   float64
Was he or she

#### FABULOUS! 

### Normalizing the data...

Let's now normalize the data, so that our analysis are accurate.

In [648]:
X = preprocessing.StandardScaler().fit(features).transform(features)
X[0:7]

array([[ 1.29210852, -0.34979314, -0.9029865 , ...,  0.        ,
         0.        ,  0.        ],
       [ 1.47613575,  0.11649234, -0.9029865 , ...,  0.        ,
         0.        ,  0.        ],
       [-0.27436718, -0.34979314, -0.9029865 , ...,  0.        ,
         0.        ,  0.        ],
       ...,
       [ 0.62332663, -0.3124903 , -0.9029865 , ...,  0.        ,
         0.        ,  0.        ],
       [-0.51225604, -0.55495875, -0.9029865 , ...,  0.        ,
         0.        ,  0.        ],
       [ 2.67006852,  0.02323524, -0.9029865 , ...,  0.        ,
         0.        ,  0.        ]])

In [649]:
y = statuses.values
y[0:7]

array([1, 1, 1, 1, 1, 1, 1])

***
## Classification

### Train-test split

### Building the model

### Training

***
## Testing accuracy

***
## Analyzing results

***
## Report

***
## Thank you!

**Author:** [Aneta Baloyan](https://www.linkedin.com/in/aneta-baloyan/)

Email: *aneta.baloyan@gmail.com*

***

May 2020

